test/TestProviderRound.spec.js

```diff
@@ -6,36 +6,95 @@ contract('ProviderRound', accounts => {
 
   let providerRound;
 
-  it('providers can register their parameters', async () => {
-    providerRound = await ProviderRound.deployed();
-    await providerRound.provider(22, 10, 1, 25, {from: accounts[0]});
-    await providerRound.provider(10, 20, 2, 35, {from: accounts[1]});
-    let firstProvider = await providerRound.providerCandidates.call(0);
-    let secondProvider = await providerRound.providerCandidates.call(1);
-    assert(firstProvider[0] == accounts[0] 
-      && firstProvider[1].toNumber() == 22
-      && firstProvider[2].toNumber() == 10
-      && firstProvider[3].toNumber() == 1
-      && firstProvider[4].toNumber() == 25
-    );
-    assert(secondProvider[0] == accounts[1] 
-      && secondProvider[1].toNumber() == 10
-      && secondProvider[2].toNumber() == 20
-      && secondProvider[3].toNumber() == 2
-      && secondProvider[4].toNumber() == 35
-    );
-  });
+  describe('provider', () => {
+
+    it("should register a provider's parameters", async () => {
+      providerRound = await ProviderRound.deployed();
+      await providerRound.provider(22, 10, 1, 25, {from: accounts[0]});
+      await providerRound.provider(10, 20, 2, 35, {from: accounts[1]});
+      let firstProvider = await providerRound.providerCandidates.call(0);
+      let secondProvider = await providerRound.providerCandidates.call(1);
+      assert(firstProvider[0] == accounts[0]
+        && firstProvider[1].toNumber() == 22
+        && firstProvider[2].toNumber() == 10
+        && firstProvider[3].toNumber() == 1
+        && firstProvider[4].toNumber() == 25
+      );
+      assert(secondProvider[0] == accounts[1]
+        && secondProvider[1].toNumber() == 10
+        && secondProvider[2].toNumber() == 20
+        && secondProvider[3].toNumber() == 2
+        && secondProvider[4].toNumber() == 35
+      );
+    });
+
+    it('should fail with invalid parameters', async() => {
+      await assertFail(
+        providerRound.provider(22, 10, -1, 25, {from: accounts[0]}),
```

Comment:
are we sure this is failing due to parameters and not the fact that it was initialized already:

https://github.com/transmute-industries/network-contracts/pull/4/files#diff-7515d8497227b639108af66aecce29b1R13

Reply:
You're right, it should be a different account

Reply:
Actually, nothing currently prevents a provider from updating their parameters so this is currently failing for the right reason.
However I will still change the account number in case we want to enforce that only one provider() call per round is allowed in the future.